docs/source/refs/reference_architecture/index.rst

```diff
@@ -0,0 +1,392 @@
+    the robot's control inputs.
+  - Event Config: Defines the set of randomizations and noisification for the agent and environment. This also includes what happens at the reset of the environment.
+  - Rewards Config: Defines the reward function for the task
+  - Terminations Config: Defines the conditions for termination of an episode or when the task
+    is completed.
+
+- You can add other optional configuration classes such as Curriculum Config for tasks that require curriculum learning and Commands Config for tasks where the input is from a controller/setpoint controls e.g. a gamepad controller.
+
+.. tip::
+
+  To learn more on how you can design your own manager-based environment, see :ref:`tutorial-create-manager-rl-env`.
+
+
+
+**Direct**
+
+.. image:: ../../_static/task-workflows/direct-based-light.svg
+    :class: only-light
+    :align: center
+    :alt: Direct-based Task Workflow
+
+.. image:: ../../_static/task-workflows/direct-based-dark.svg
+    :class: only-dark
+    :align: center
+    :alt: Direct-based Task Workflow
+
+In this workflow, you implement a single class that is responsible for computing observations, applying actions, and computing rewards. This workflow allows for direct control of the environment logic.
+
+.. tip::
+  To learn more on how you can design your own direct environment, see :ref:`tutorial-create-direct-rl-env`.
+
+Users can choose from Isaac Lab’s large suite of pre-configured environments or users can define
+their own environments. For more technical information about the two workflows, please see the
+`documentation <https://isaac-sim.github.io/IsaacLab/main/source/overview/core-concepts/task_workflows.html>`__.
+
+
+In addition to designing the RL task, you will need to design your agent’s model, the neural
+network policy and value function. To train the RL agent to solve the task, you need to define
+the hyperparameters such as number of epochs, learning rate, etc. for training and the
+policy/value model architecture. This is defined in the training configuration file specific
+to the RL library you want to use. Examples are created under the agent's folder in each task directory.
+See an example of `RSL-RL <https://github.com/isaac-sim/IsaacLab/blob/main/source/extensions/omni.isaac.lab_tasks/omni/isaac/lab_tasks/manager_based/locomotion/velocity/config/anymal_b/agents/rsl_rl_ppo_cfg.py>`__ for Anymal-B.
+
+
+**Component 4 - Register with Gymnasium**
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The next step is to register the environments with the gymnasium registry.
+Registration is a way to make the environment accessible and reusable across different
+RL algorithms and experiments. This is common in the RL community. Follow the tutorial on
+`Registering an Environment <https://isaac-sim.github.io/IsaacLab/main/source/tutorials/03_envs/register_rl_env_gym.html>`__ to learn more about how to register in your own environment.
+
+**Component 5 - Environment Wrapping**
```

Comment:
this seems like a rather optional step, compared to the ones above. maybe we should differentiate it somehow so that users are not confused about having to do this.

Reply:
Rephrased here: 

**Component 4 - Register with Gymnasium (Optional)**
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

This step is optional because the user can manually create an instance of the environment by importing the environment class and its configuration class. However, you can also register the environments with the gymnasium registry.

Reply:
I meant this for section 5 Environment Wrapping. I think registering with gymnasium should still be required if you want to run with their task name?

Reply:
Yes

Reply:
Environment Wrapping is already with the tone of optional though. But they need the RL libraries wrappers, except that it is already done